tasks/connectors/github_dependabot/github_dependabot.rb

```diff
@@ -0,0 +1,180 @@
+            { name: "kenna_connector_id",
+              type: "integer",
+              required: false,
+              default: nil,
+              description: "If set, we'll try to upload to this connector" },
+            { name: "output_directory",
+              type: "filename",
+              required: false,
+              default: "output/github_dependabot",
+              description: "If set, will write a file upon completion. Path is relative to #{$basedir}" }
+          ]
+        }
+      end
+
+      # Github graphql configuration
+
+      http = GraphQL::Client::HTTP.new("https://api.github.com/graphql") do
+        def headers(_context)
+          { "Authorization": "Bearer #{GITHUB_DEPENDABOT_TOKEN}" }
+        end
+      end
+
+      Schema = GraphQL::Client.load_schema(http)
+
+      Client = GraphQL::Client.new(schema: Schema, execute: http)
+
+      # Graphql query
+
+      SECURITY_ADVISORY_QUERY = Client.parse <<-'GRAPHQL'
+        query {
+          organization(login: "KennaSecurity") {
+            repositories(orderBy: {field: UPDATED_AT, direction: DESC}, first: 50) {
+              nodes {
+              name
+                vulnerabilityAlerts(last: 50) {
+                  nodes {
+                    id
+                    securityAdvisory {
+                      description
+                      cvss {
+                        score
+                      }
+                      severity
+                      identifiers {
+                        type
+                        value
+                      }
+                      summary
+                      vulnerabilities(last: 50) {
+                        nodes {
+                          package {
+                            name
+                          }
+                          severity
+                          firstPatchedVersion {
+                            identifier
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      GRAPHQL
+
+      def self.security_advisories
+        GithubDependabot::Client.query(SECURITY_ADVISORY_QUERY)
+      end
+
+      def run(opts)
+        super # opts -> @options
+
+        @github_access_token = @options[:github_access_token]
+        @kenna_api_host = @options[:kenna_api_host]
+        @kenna_api_key = @options[:kenna_api_key]
+        @kenna_connector_id = @options[:kenna_connector_id]
+
+        repos = GithubDependabot.security_advisories.original_hash["data"]["organization"]["repositories"]["nodes"] # get the response from graphql query
+
+        repo_map = repos.each_with_object({}) do |repo, h|
+          advisories = repo["vulnerabilityAlerts"]["nodes"].map { |alert| alert["securityAdvisory"] }
+          h[repo["name"]] = advisories.reject { |ad| ad["identifiers"].last.value?("GHSA") }
+        end
+
+        kdi_hash = {
```

Comment:
please review and use the help methods to complete the functions from 116 to 150

create_kdi_asset_vuln(asset, vulnerability_alerts_for(repo))
create_kdi_vuln_def(vulnerability_definitions(repo_map))

and then 

kdi_upload(@output_directory, "sample_scan_#{scan_id}_report_#{pos}.json", @kenna_connector_id, @kenna_api_host, @kenna_api_key, @skip_autoclose, @retries, @kdi_version)

let me know if you have any questions.

Reply:
@imishraSecurview